timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -216,19 +133,19 @@ void __device__ v_nonbonded_unified(
     unsigned long long *__restrict__ du_dp,
     unsigned long long *__restrict__ u_buffer) {
 
-    int tile_idx = blockIdx.x;
-
-    RealType box_x = box[0 * 3 + 0];
-    RealType box_y = box[1 * 3 + 1];
-    RealType box_z = box[2 * 3 + 2];
+    RealType box_x = shared_box.x;
+    RealType box_y = shared_box.y;
+    RealType box_z = shared_box.z;
 
-    RealType inv_box_x = 1 / box_x;
-    RealType inv_box_y = 1 / box_y;
-    RealType inv_box_z = 1 / box_z;
+    RealType inv_box_x = shared_box.inv_x;
```

Comment:
I'm really surprised that it's faster to load 1/box from box_cache than to compute it directly?